Remove advocacy site from pegasus

kelbyhawn opened the pull request:
Removes the `pegasus/sites.v3/advocacy.code.org` directory from pegasus. This site has been moved over to Leadpages and still lives at https://advocacy.code.org.

There are other areas related to the advocacy.code.org domain integrated throughout the codebase that I tried removing, but it resulted in hard to diagnose failing tests. If anyone has insight into what else we can remove I can do that in a separate PR. 

## Links
- Jira ticket: [ACQ-2052](https://codedotorg.atlassian.net/browse/ACQ-2052)

## Testing story
I'm not entirely sure how to test this, but hoping Drone will catch things.

## Followup
- Remove advocacy.code.org from pegasus router - Jira ticket: [ACQ-2087](https://codedotorg.atlassian.net/browse/ACQ-2087)
- Remove advocacy.code.org references from the codebase - Jira ticket: [ACQ-2086](https://codedotorg.atlassian.net/browse/ACQ-2086)

kelbyhawn commented:
@cat5inthecradle @bethanyaconnor there's something happening here too, but I'm not sure how (or if) I should remove it: https://github.com/code-dot-org/code-dot-org/blob/5efa1b3863199c55727717f094fdba7d57771576/lib/cdo/cloud_formation/cdo_app.rb#L190

bethanyaconnor reviewed (approved):
Can you make a ticket to remove other references? (And maybe link to the commits you reverted?) A lot of them might need infra or platform support.

Overall I think this is a good set of deletions, though I might add a couple more things, if Drone cooperates:
- removing from the [dropbox sync](bin/cron/sync_dropbox) (not sure how it'll go if someone did try to make a change in Dropbox)
- removing the [UI test](dashboard/test/ui/features/other_sites.feature)
- removing references from the [pegasus router](pegasus/router.rb) -- though I might do that in a separate PR in case it create surprise issues

kelbyhawn commented:
> Can you make a ticket to remove other references? (And maybe link to the commits you reverted?) A lot of them might need infra or platform support.

@bethanyaconnor done! Added ticket links to the [Followup](https://github.com/code-dot-org/code-dot-org/pull/59844#issue-2414344166) section above.

I also removed the Dropbox ref and all of the tests (except for one in pegasus that I'll remove w/ the router stuff). Drone is passing w/ these changes so I feel better about merging 🎉
